### PR TITLE
[#12] I'm not sure it's works

### PR DESCRIPTION
**Pull request by wapgear**
About issue #12

### CODE REVIEW

File: handler.js [hunk 1]
```diff
@@ -59,7 +59,11 @@ export const nofap = async (event, context, callback) => {
         case 'top':
             let top1 = await nfService.getTop()
             callback(null, topNF(top1))
-            break
+            break;
+
+        case 'list':
+            callback(null, participantsList(nfService.getList()));
```
Comment (buzdyk) [inline handler.js]: lets remove `;` in the end of the line

File: handler.js [hunk 2]
```diff
@@ -59,7 +59,11 @@ export const nofap = async (event, context, callback) => {
         case 'top':
             let top1 = await nfService.getTop()
             callback(null, topNF(top1))
-            break
+            break;
+
+        case 'list':
```
Comment (buzdyk) [inline handler.js]: `list` may mean several things, e.g.

- list of my nofaps
- list of participants

Let's rename the command to `participants`

File: helpers/responses.js [hunk 3]
```diff
@@ -85,6 +85,13 @@ export const topNF = stats => {
     }
 }
 
+export const participantsList = users => {
+    const text = `
```
Comment (buzdyk) [inline helpers/responses.js]: let's return text with an explicit "ephemeral" response_type, like:

`return {response_type: 'ephemeral', text}`

File: helpers/responses.js [hunk 4]
```diff
@@ -132,6 +139,7 @@ NoFap slash commands:
 - _/nofap oopsie (optional comment)_ - ends your NoFap. Hint: do not start a new NoFap right away. Have some time and think of your motivation
 - _/nofap stats_ - displays your info (privately)
 - _/nofap top_ – displays top participants
+- _/nofap list_ – displays all participants
```
Comment (buzdyk) [inline helpers/responses.js]: list -> participants

File: db/nofap-service.js [hunk 5]
```diff
@@ -48,6 +48,10 @@ export default class NofapService {
         return items.length ? items[0] : null
     }
 
+    async getList() {
+        return await this.getActive().sort(i => i.start);
```
Comment (buzdyk) [inline db/nofap-service.js]: lets remove ; in the end of the line

File: helpers/responses.js (hunk 3 above)
Comment (wapgear) [reply helpers/responses.js]: whoops

**Comment by wapgear** [comment]
Hey @buzdyk please take another view

**Pull request merged**